src/vanilla/ModelNameComparer.cs

```diff
@@ -0,0 +1,24 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Collections.Generic;
+using AutoRest.Core.Model;
+
+namespace AutoRest.Java
+{
+    public class ModelNameComparer : IEqualityComparer<ModelType>
```

Comment:
Did this class accidentally slip into this pull request?

Reply:
Nope, but when the XML serialization stuff is consolidated it might go back to being an inner class.